builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -275,9 +277,14 @@ func resourceAwsS3BucketObjectRead(d *schema.ResourceData, meta interface{}) err
 			Key:    aws.String(key),
 		})
 	if err != nil {
-		return fmt.Errorf("Failed to get object tags (bucket: %s, key: %s): %s", bucket, key, err)
+		// Treat the inability to get object tags in a restricted regions as a
+		// soft error.
+		if !restricted {
+			return fmt.Errorf("Failed to get object tags (bucket: %s, key: %s): %s", bucket, key, err)
```

Comment:
It would allow them to be set, but would then fail to refresh them, I think? I slightly favor being explicit about what's supported and then explicitly removing both of these codepaths in a future release if and when the other partitions get this support, but I don't feel strongly about it so I will cede to you both (and your greater experience with dealing with GovCloud) if you think silently ignoring specified tags is the better approach.